toolkit/smart-contracts/offchain/src/sign_tx.rs

```diff
@@ -0,0 +1,69 @@
+use crate::{cardano_keys::CardanoPaymentSigningKey, csl::transaction_from_bytes};
+use cardano_serialization_lib::{Ed25519Signature, Vkey, Vkeywitness};
+
+pub fn sign_tx(tx: Vec<u8>, payment_key: &CardanoPaymentSigningKey) -> anyhow::Result<Vec<u8>> {
```

Comment:
We already do sign some transactions in integration tests. It would be nice to use `sign_tx` and feed the result to `assemble_tx` (sorry for not 100% precise names - I'm at different branch now).